### PR TITLE
Setup Firebase (with authentication and Redux)

### LINKED ISSUE #6: Setup firebase

### LINKED ISSUE #13: Create a Redux store

### PR DESCRIPTION
**Pull request by Atema**
Resolves #6, resolves #13.

It adds dependencies for `firebase`, `redux`, and for integration between them and react (`react-redux`, `react-redux-firebase` (RRF)).

Firebase is supposed to be used through RRF. Login, Signup and Logout were added for authentication through Firebase.

### CODE REVIEW

File: src/Components/SignupForm/SignupForm.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+import { useFirebase } from "react-redux-firebase";
+
+import SignupFormView from "./SignupFormView";
+
+const SignupForm = function () {
+  const firebase = useFirebase();
+  const [error, setError] = useState(null);
+
+  const handleSignup = (email, password, passwordConfirm) => {
```
Comment (vanFrZy) [inline src/Components/SignupForm/SignupForm.js:10]: Maybe we can add some check to see if the email is valid? Or is this handled ny firebase itself

Comment (Atema) [reply src/Components/SignupForm/SignupForm.js:10]: Yep, it's handled by firebase already :)

File: src/Pages/LoginPage/LoginPage.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { Navigate } from "react-router";
+import useAuthentication from "Services/firebase/useAuthentication";
+import useTitle from "Utils/useTitle";
+
+const LoginPage = function () {
+  useTitle("Bragi 3000 - Login");
+
+  const { authReady, loggedIn } = useAuthentication();
+
+  if (!authReady)
```
Comment (vanFrZy) [inline src/Pages/LoginPage/LoginPage.js:11]: Duplicated code from SignupPage. However for this small amount of code I don't think we need to extract it.

**Review by vanFrZy — COMMENTED** [review]
Well done! I'm only not really sure what the src/Store/ folder is used for, could you explain?

**Comment by Atema** [comment]
> Well done! I'm only not really sure what the src/Store/ folder is used for, could you explain?

Thanks! The store folder is for the Redux store, and usually contains all reducers (functions related to data persistence). Most of it is managed by `react-redux-firebase`, but it could contain functions for getting and setting data in firebase (via redux).

File: src/Pages/LoginPage/LoginPage.js (hunk 2 above)
Comment (Atema) [reply src/Pages/LoginPage/LoginPage.js:11]: Yeah, this usage (the page should only be accessible when you're *not* logged in) is quite rare. I first wanted to create some `PrivateRoute` component (for routes only accessible when you *are* logged in), but even that will only be used in the `App` page (I think), so in the end I decided to do it like this, with the `useAuthentication` hook abstracting away some of the logic.

**Review by vanFrZy — APPROVED** [review]

**Pull request merged**